### ISSUE TITLE
Getting `Illegal instruction` when compiling with mpiicpc and mpiicc

### ISSUE THREAD

**Issue by BJWiley233** [plumed/plumed2#904]
I am compiling on Ubuntu 20.04 machine with the following after `git clone https://github.com/plumed/plumed2.git`:

```
## source intel
. /opt/intel/oneapi/setvars.sh -arch intel64 -platform linux
## compile
./configure --prefix=/usr/local CXX=mpiicpc CC=mpiicc && make -j 16
```

Everything compiles however when it gets near the final step and tries to call compiled plumed program in the json library [here](https://github.com/plumed/plumed2/blob/master/json/Makefile#L26) it says `Illegal instruction`

```
../src/lib/plumed gen_json --actions action_list > syntax.json
Illegal instruction
```

It compiled and worked fine without mpi but I need mpi, `./configure --prefix=/usr/local CXX=icpc CC=icc`.  This may not work with gromacs in future as gromacs is doing away with icc/icpc.

UPDATE: I am seeing this WARNING line during configure
```
configure: **** PLUMED will be compiled using MPI
configure: WARNING: plumed executable will not run on this machine
configure: WARNING: unless you invoke it as 'plumed --no-mpi'
```
So maybe just need to edit the Makefile inside json folder since I am building it in a docker to run on an LSF HPC environment.

**Comment by GiovanniBussi** [comment]
Are you cross compiling? I mean: are you using flags that would build an executable compatible with another architecture? I think this is the only expected case where this could happen.

**Comment by GiovanniBussi** [comment]
Another comment. If you go up in the log, you should see a check like `whether a program can be run on this machine`. In theory, if this is `no`, building the `syntax.json` file should be skipped. This is what we should do when e.g. cross compiling.

However, it is possible that you have compiler flags so that:
- the simple executable tested during configure can run
- but the plumed executable cannot run because it includes some incompatible instructions

A possible hack would be to manually (or with a script) replace the following line:
```
program_can_run=yes
```
with
```
program_can_run=no
```
in `Makefile.conf`, *after* configure, *before* make

**Comment by GiovanniBussi** [comment]
Regarding this:

>It compiled and worked fine without mpi but I need mpi, ./configure --prefix=/usr/local CXX=icpc CC=icc. This may not work with gromacs in future as gromacs is doing away with icc/icpc.

I am not sure I understand. As far as I can guess, if GROMACS stops supporting `icpc`, it will also stop supporting `mpiicpc`. So I don't think this is related to your issue.

My guess is that GROMACS will ask people to use the latest intel compilers (`icx` etc).

**Comment by GiovanniBussi** [comment]
@BJWiley233 sorry maybe I noticed an error.

In file `json/Makefile`, try replacing:
```
../src/lib/plumed gen_json --actions action_list > syntax.json
```
with
```
../src/lib/plumed gen_json --no-mpi --actions action_list > syntax.json
```

Let me know if this fixes the problem. Thanks

**Comment by BJWiley233** [comment]
That makes sense for the icpc/mpiicpc. I guess intel's latest comilers will provide mpi versions.  I am not cross-compiling, I used the same mpiicpc compiler for both plumed and Gromacs.  And so I already added the `--no-mpi` flag for creating the the `syntax.json` file.  Maybe I don't even need this syntax file as you are stating however?  I could just delete the json folder completely maybe?  Or does it not hurt anything but having it even if we are not cross-compiling?
